src/System.Device.Gpio/System/Device/Gpio/GpioController.cs

```diff
@@ -310,11 +333,17 @@ public void UnregisterCallbackForPinValueChangedEvent(int pinNumber, PinChangeEv
             _driver.RemoveCallbackForPinValueChangedEvent(logicalPinNumber, callback);
         }
 
-        private void Dispose(bool disposing)
+        /// <summary>
+        /// Disposes this instance and closes all open pins associated with this controller.
+        /// </summary>
+        /// <param name="disposing">True to dispose all instances, false to dispose only unmanaged resources</param>
+        protected virtual void Dispose(bool disposing)
         {
-            foreach (int pin in _openPins)
+            var tempList = new List<int>(_openPins); // Because ClosePin modifies this list
```

Comment:
should you instead create ClosePinInternal or similar method which doesn't modify the list?

Reply:
Not so easy, because ClosePin is virtual and any inheritors shouldn't have to worry about a distinction there.